addons/delivery/models/stock_picking.py

```diff
@@ -97,6 +86,10 @@ def _compute_carrier_tracking_url(self):
         for picking in self:
             picking.carrier_tracking_url = picking.carrier_id.get_tracking_link(picking) if picking.carrier_id and picking.carrier_tracking_ref else False
 
+    def _compute_number_of_package(self):
+        for picking in self:
+            picking.number_of_packages = len(self.move_line_ids.mapped('result_package_id.id'))
+
```

Comment:
If there's a chance that pickings have a large number of packages, might be worth [using a read group](https://github.com/odoo/odoo/pull/21198) preemptively.